cxx_argp_parser.h

```diff
@@ -53,19 +53,43 @@ class parser
 		};
 	}
 
-	/* for integers */
+	/* for signed integers */
 	template <typename T>
 	typename std::enable_if<
-	    std::numeric_limits<T>::is_integer,
+	    std::numeric_limits<T>::is_integer && std::numeric_limits<T>::is_signed,
 	    std::function<bool(const char *)>>::type
 	make_check_function(T &x)
 	{
 		const auto px = &x;
 		return [px](const char *arg) {
 			char *end;
-			*px = strtol(arg, &end, 0);
+			long long int value = strtoll(arg, &end, 0);
 			if (*end != '\0') // not all of the string has been consumed
 				return false;
+			if (value < std::numeric_limits<T>::min() ||
```

Comment:
Hmm, maybe merging both function won't be possible because in the case of unsigned this would create a warning maybe.

Reply:
not sure about warnings, but I'm pretty sure a compiler would be clever enough to optimise that comparison out for unsigned types